File: data_structures/queue/php/queue.php
<?php

/**
 * Class Queue by Bennett Treptow
 * Pretty trivial implementation
 * PHP arrays and array_shift do the hard work
 */
class Queue {
    /**
     * @var array
     */
    protected $_data = [];

    /**
     * Queue constructor.
     * @param null $data
     */
    public function __construct($data = null){
        if(!is_null($data)){
            foreach($data as $item){
                $this->enqueue($item);
            }
        }
    }

    /**
     * @return int
     */
    public function size(){
        return count($this->_data);
    }

    /**
     * @param $item
     */
    public function enqueue($item){
        $this->_data[] = $item;
    }

    /**
     * @return mixed
     * @throws Exception
     */
    public function dequeue(){
        if($this->size() > 0){
            return array_shift($this->_data);
        } else {
            throw new Exception('Can\'t dequeue an empty queue!');
        }
    }
}

$queue = new Queue(['1','2','3','4']);
$queue->dequeue(); //1
$queue->enqueue('5'); //2,3,4,5
?>

File: data_structures/queue/go/queue.go
package main

import "fmt"

type queue struct {
    data []interface{}
}

func NewQueue() queue {
    return queue{nil}
}

func (q *queue) Enqueue(value interface{}) {
    q.data = append(q.data, value)
}

func (q *queue) Dequeue() (interface{}) {
    value := q.data[0]
    q.data = q.data[1: len(q.data)]
    return value
}

func main() {
    q := NewQueue()
    q.Enqueue(10)
    q.Enqueue("Hello")
    q.Enqueue(20)
    q.Enqueue(10)
    q.Enqueue("Hello")
    q.Enqueue(20)
    q.Enqueue(10)
    q.Enqueue("Hello")
    q.Enqueue(20)
    q.Enqueue(10)
    q.Enqueue("Hello")
    q.Enqueue(20)
    q.Enqueue(10)
    q.Enqueue("Hello")
    q.Enqueue(20)
    q.Enqueue(10)
    q.Enqueue("Hello")
    q.Enqueue(20)
    fmt.Println(q)
    fmt.Println(q.Dequeue())
    fmt.Println(q.Dequeue())
    fmt.Println(q)
}
